pkg/client/automation/client.go

```diff
@@ -109,36 +110,53 @@ func WithClientRequestLimiter(limiter *concurrency.Limiter) func(client *Client)
 }
 
 // List returns all automation objects
-func (a Client) List(resourceType ResourceType) (result *ListResponse, err error) {
+func (a Client) List(resourceType ResourceType) (result []Response, err error) {
 	a.limiter.ExecuteBlocking(func() {
 		result, err = a.list(resourceType)
 	})
 	return
 }
 
-func (a Client) list(resourceType ResourceType) (*ListResponse, error) {
-	// try to get the list of resources
-	resp, err := rest.Get(a.client, a.url+a.resources[resourceType].Path)
-	if err != nil {
-		return nil, fmt.Errorf("unable to list automation resources: %w", err)
-	}
+func (a Client) list(resourceType ResourceType) ([]Response, error) {
+	var retVal []Response
+	var result ListResponse
+	result.Count = 1
 
-	// handle http error
-	if !resp.IsSuccess() {
-		return nil, ResponseErr{
-			StatusCode: resp.StatusCode,
-			Message:    "Failed to list automation objects",
-			Data:       resp.Body,
+	for len(retVal) < result.Count {
+
+		u, err := internal.NextPageURL(a.url, a.resources[resourceType].Path, len(retVal))
```

Comment:
nit: I would give the sub-package a more speaking name, like `url`.

Or simply keep the `url.go` and `url_test.go` files in the `client/automation` package and un-export the method if it's only meant to be used here.